components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/model/AccessToken.java

```diff
@@ -0,0 +1,256 @@
+                if (claim.getName().equals(name)) {
+                    return claim;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    public AccessToken.Builder copy() {
+
+        return new Builder()
+                .tokenType(this.tokenType)
+                .scopes(new ArrayList<>(this.scopes))
+                .claims(this.claims.stream().map(Claim::copy).collect(Collectors.toList()));
+    }
+
+    /**
+     * Enum for standard claim names.
+     */
+    public enum ClaimNames {
+
+        SUB("sub"),
+        ISS("iss"),
+        AUD("aud"),
+        CLIENT_ID("client_id"),
+        AUTHORIZED_USER_TYPE("aut"),
+        EXPIRES_IN("expires_in"),
+
+        TOKEN_BINDING_REF("binding_ref"),
+        TOKEN_BINDING_TYPE("binding_type"),
+        SUBJECT_TYPE("subject_type");
+
+        private final String name;
+
+        ClaimNames(String name) {
+
+            this.name = name;
+        }
+
+        public static boolean contains(String name) {
+
+            return Arrays.stream(ClaimNames.values())
+                    .anyMatch(claimName -> claimName.name.equals(name));
+        }
+
+        public String getName() {
+
+            return name;
+        }
+    }
+
+    /**
+     * Model class for claims.
+     */
+    public static class Claim {
+
+        private String name;
+        private Object value;
+
+        public Claim() {
+
+        }
+
+        public Claim(String name, Object value) {
+
+            this.name = name;
+            this.value = value;
+        }
+
+        public String getName() {
+
+            return name;
+        }
+
+        public void setName(String name) {
+
+            this.name = name;
+        }
+
+        public Object getValue() {
+
+            return value;
+        }
+
+        public void setValue(Object value) {
+
+            this.value = value;
+        }
+
+        public Claim copy() {
+
+            return new Claim(this.name, deepCopyValue(this.value));
+        }
+
+        private Object deepCopyValue(Object value) {
+
+            if (value instanceof List) {
+                List<?> originalList = (List<?>) value;
+                List<Object> copiedList = new ArrayList<>();
+                for (Object item : originalList) {
+                    copiedList.add(deepCopyValue(item)); // Recursive for nested lists
+                }
+                return copiedList;
+            } else if (value instanceof Map) {
+                Map<?, ?> originalMap = (Map<?, ?>) value;
+                Map<Object, Object> copiedMap = new HashMap<>();
+                for (Map.Entry<?, ?> entry : originalMap.entrySet()) {
+                    copiedMap.put(entry.getKey(), deepCopyValue(entry.getValue())); // Recursive for nested maps
+                }
+                return copiedMap;
+            } else if (value.getClass().isArray()) {
+                return Arrays.copyOf((Object[]) value, ((Object[]) value).length);
+            } else {
```

Comment:
This else is redundant. We can just add  "// For immutable types or types not requiring deep copy
                return value; " without specifying the else block

Reply:
Since there is an else if, this else is needed. 
It's not redundant and the IDE doesn't report.